.github/workflows/publish-snapshot.yml

```diff
@@ -37,7 +37,7 @@ jobs:
       - uses: actions/setup-java@v4
         with:
           distribution: zulu
-          java-version: 11
+          java-version: 21
```

Comment:
Spark 3.4 doesn't compile with 21. I think 17 works for all versions and modules.